aws-logging-cloudwatch/src/androidTest/kotlin/com/amplifyframework/logging/cloudwatch/db/CloudWatchLoggingDatabaseInstrumentationTest.kt

```diff
@@ -0,0 +1,155 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amplifyframework.logging.cloudwatch.db
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.amplifyframework.logging.cloudwatch.models.CloudWatchLogEvent
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.Assert.*
+
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.time.Instant
+
+@RunWith(AndroidJUnit4::class)
+@OptIn(ExperimentalCoroutinesApi::class)
+class CloudWatchLoggingDatabaseInstrumentationTest {
+    private val context = InstrumentationRegistry.getInstrumentation().context
+    private val testCoroutine = UnconfinedTestDispatcher()
+    private val loggingDbClass = CloudWatchLoggingDatabase(context, testCoroutine)
+    private val database by lazy {
+        System.loadLibrary("sqlcipher")
+        val passPhraseGetter = loggingDbClass.javaClass.getDeclaredMethod("getDatabasePassphrase", String::class.java)
+        passPhraseGetter.isAccessible = true
+        CloudWatchDatabaseHelper(context).getWritableDatabase(passPhraseGetter.invoke(loggingDbClass) as String)
+    }
+
+    private val testTimestamp1 = Instant.now().epochSecond
+    private val testTimestamp2 = Instant.now().epochSecond + 300
+    private val testTimestamp3 = Instant.now().epochSecond + 600
+    private val testTimestamp4 = Instant.now().epochSecond + 900
+    private val testCloudWatchLogEvent1 = CloudWatchLogEvent(testTimestamp1, "Customer Obsession")
+    private val testCloudWatchLogEvent2 = CloudWatchLogEvent(testTimestamp2, "Ownership")
+    private val testCloudWatchLogEvent3 = CloudWatchLogEvent(testTimestamp3, "Bias for Action")
+    private val testCloudWatchLogEvent4 = CloudWatchLogEvent(testTimestamp4, "Frugality")
+
+    @Before
+    fun setUp() = runTest {
+        Dispatchers.setMain(testCoroutine)
+    }
+
+    @After
+    fun tearDown() = runTest {
+        // Clear the db used for testing
+        loggingDbClass.clearDatabase()
+    }
+
+    /**
+     * This test verifies if the CloudWatch Logs are saved as intended by passing
+     * CloudWatchLogEvent to saveLogEvent() method
+     */
+    @Test
+    fun saveLogEvent_Saves_Two_Logs_After_Two_Calls() = runTest {
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent1)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent2)
+
+        val savedLogs = loggingDbClass.queryAllEvents()
+
+        assertNotNull(savedLogs)
+        assertEquals(2, savedLogs.size)
+        assertEquals("Customer Obsession", savedLogs[0].message)
+        assertEquals(testTimestamp2, savedLogs[1].timestamp)
```

Comment:
Trying to get us to use [Kotest Assertions](https://kotest.io/docs/assertions/assertions.html) instead of basic junit. Add a dependency `androidTestImplementation(libs.test.kotest.assertions)`.

The way I would write this test is to add a customer matcher:

```kotlin
private infix fun LogEvent.shouldMatchEvent(event: CloudWatchLogEvent) = this should Matcher {
        MatcherResult(
            it.message == event.message && it.timestamp == event.timestamp,
            { "event $it should match $event but it does not" },
            { "event $it should not match $event but it does" }
        )
    }
```

and then it becomes easy to match lists of `LogEvent` against the expected values. This line is now a complete assertion that the size of the list plus all of the values exactly match:

```kotlin
savedLogs shouldMatchEach listOf(
    { it shouldMatchEvent testCloudWatchLogEvent1 },
    { it shouldMatchEvent testCloudWatchLogEvent2 }
)
```

You can even go a step further and make a custom assertion for the list:

```kotlin
private fun List<LogEvent>.shouldMatchEvents(vararg events: CloudWatchLogEvent) =
        this shouldMatchEach events.map { event -> { it shouldMatchEvent event } }
```

And now your assertion is:

```kotlin
savedLogs.shouldMatchEvents(testCloudWatchLogEvent1, testCloudWatchLogEvent2)
```

Which you can reuse in each test.

Reply:
Made changes! Will apply these for future tests as well 👍